src/main/java/roomescape/controller/api/ReservationController.java

```diff
@@ -0,0 +1,70 @@
+package roomescape.controller.api;
+
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import roomescape.dto.ReservationRequest;
+import roomescape.dto.ReservationResponse;
+import roomescape.exception.CannotCreatedException;
+import roomescape.exception.DuplicateReservationException;
+import roomescape.exception.NotCorrectDateTimeException;
+import roomescape.exception.ThemeDoesNotExistException;
+import roomescape.exception.TimeDoesNotExistException;
+import roomescape.service.ReservationService;
+
+@RestController
+@RequestMapping("/reservations")
+public class ReservationController {
+
+    private final ReservationService reservationService;
+
+    private ReservationController(ReservationService reservationService) {
+        this.reservationService = reservationService;
+    }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public ReservationResponse createReservation(@RequestBody ReservationRequest request) {
+        return ReservationResponse.from(reservationService.addReservationAfterNow(request));
+    }
+
+    @PostMapping("/admin")
+    @ResponseStatus(HttpStatus.CREATED)
+    public ReservationResponse createAdminReservation(@RequestBody ReservationRequest request) {
+        return ReservationResponse.from(reservationService.addReservation(request));
+    }
+
+    @GetMapping
+    public List<ReservationResponse> readReservations() {
+        return reservationService.findAllReservations().stream()
+            .map(ReservationResponse::from)
+            .toList();
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteReservation(@PathVariable Long id) {
+        reservationService.removeReservation(id);
+    }
+
+    @ExceptionHandler(value = {TimeDoesNotExistException.class, ThemeDoesNotExistException.class,
+        NotCorrectDateTimeException.class})
```

Comment:
아래 처럼 변경하는 것과 차이가 없을 것 같은데, 모두 적어준 이유가 있을까요?
```suggestion
     @ExceptionHandler(value = {CannotCreatedException.class})
```

Reply:
구현 과정에서 예외 클래스가 추가되면서 마지막에 관련 부분을 반영하는 것을 깜빡했던 것 같습니다.. 🥲
CustomException을 도입하면서, 모든 커스텀 예외를 GlobalExceptionHandler의 하나의 메서드에서 일관되게 처리할 수 있도록 예외 처리 방식을 통합했습니다!